fedimint-bip39/src/lib.rs

```diff
@@ -0,0 +1,48 @@
+//! BIP39 client secret support crate
+
+use std::io::{Read, Write};
+
+use fedimint_client::derivable_secret::DerivableSecret;
+use fedimint_client::secret::RootSecretStrategy;
+use fedimint_core::encoding::{Decodable, DecodeError, Encodable};
+use rand::{CryptoRng, RngCore};
+
+/// BIP39 root secret encoding strategy allowing retrieval of the seed phrase.
+#[derive(Debug)]
+pub struct Bip39RootSecretStrategy<const WORD_COUNT: usize = 12>;
+
+impl<const WORD_COUNT: usize> RootSecretStrategy for Bip39RootSecretStrategy<WORD_COUNT> {
```

Comment:
I can't look deeper into it RN, but couldn't it operate on just `String` (whole phrase), instead of splitting things in words in type system etc.?